platform/cavium/cavm_platform_modules/Makefile

```diff
@@ -0,0 +1,27 @@
+.ONESHELL:
+SHELL = /bin/bash
+.SHELLFLAGS += -e
+
+MAIN_TARGET = cavm_platform_modules.deb
+DEB_BUILD_DIR = cavm-platform-modules-deb
+KVERSION = linux-3.16.36
+
+$(addprefix $(DEST)/, $(MAIN_TARGET)): $(DEST)/% :
+	# get sources
+	rm -rf SONiC
+	git clone https://github.com/edge-core/SONiC.git
+
+	# build
+	pushd SONiC/AS7512-32X/module/
+
+	KERNEL_SRC=/sonic/src/sonic-linux-kernel/$(KVERSION)/debian/build/build_amd64_none_amd64 make
```

Comment:
Every version of linux ships linux-headers that can be installed for modules compilation (linux-headers-3.16.0-4-amd64_3.16.36-1+deb8u2_amd64.deb and linux-headers-3.16.0-4-common_3.16.36-1+deb8u2_amd64.deb for current linux version). These .deb packages contain headers and Makefiles needed in order to compile external modules and are always installed into /lib/modules/linux-headers-$(uname -r)/.

So to compile modules KERNEL_SRC is always equal to /lib/modules/$(KVERSION)/build and KVERSION is always equal to kernel release (3.16.0-4-amd64 in our case).